src/sys/unix/tcp.rs

```diff
@@ -254,3 +307,19 @@ impl AsRawFd for TcpListener {
         self.inner.as_raw_fd()
     }
 }
+
+// Implementation taken from the Rust standard library.
+// Copyright 2015 The Rust Project Developers.
```

Comment:
So.... licensing / copyright fun :) where did you get the copyright snippet from? Looking [here](https://github.com/rust-lang/rust/blob/master/LICENSE-MIT), I didn't see anything.

Since mio is MIT and rust is MIT, there is no problems copying as long as we respect the [MIT license](https://github.com/rust-lang/rust/blob/master/LICENSE-MIT).

Key bit:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Except, Rust removed the copyright notice [here](https://github.com/rust-lang/rust/commit/2a8807e889a43c6b89eb6f2736907afa87ae592f#diff-7f0bdbc0a0545831259b66259ac6b604), which feels incorrect (of course, I am not a lawyer).

However, two things:

- You only need to maintain the license text for copies of *substantial* bits, and I don't think this counts (small and "obvious").
- The code is from **before** the copyright line was removed from the license.

So, I would say we should do one of two things:
- Remove this comment
- Copy [this](https://github.com/rust-lang/rust/blob/8c96f37a98a19444e9c8a82b86732c742b65c8ce/LICENSE-MIT) file verbatim at the end of our license file.

Reply:
I got the function from https://github.com/rust-lang/rust/blob/97b1128589fdaa786a7cf65c5a6ff7ed37a1d2f3/src/libstd/net/addr.rs#L588-L599. But I could change the code to drop the function.

Reply:
Copying is fine, i would just use one of the two strategies above.

Reply:
I'll remove the copyright and continue with the other places net2 is used.